stripe/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -215,6 +214,18 @@ internal class PaymentSheetViewModel internal constructor(
             StripeIntentResult.Outcome.SUCCEEDED -> {
                 eventReporter.onPaymentSuccess(selection.value)
 
+                // SavedSelection needs to happen after new cards have been saved.
```

Comment:
Is this true when "save card" is unchcked?

Reply:
It could happen before, but I didn't think it was required to?  Are there scenarios I haven't thought through?

Reply:
"saved" has become an overloaded term so I'm making sure I'm following the logic. I think what you have here may be right.

Reply:
yeah, so I can try to explain some more, so the item being saved here, is the one to default preferences repository which is to remember the last payment selection used.   So I move the same from happening when you first click on checkout until after that happens and after the confirmAndAuth result is processed.